src/libraries/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/Formatters/Binary/BinaryObjectReader.cs

```diff
@@ -75,6 +75,21 @@ internal ObjectReader(Stream stream, ISurrogateSelector? selector, StreamingCont
             _formatterEnums = formatterEnums;
         }
 
+        private static IDisposable? StartDeserialization()
+        {
+            MethodInfo? targetMethod = typeof(SerializationInfo).GetMethod(
+                "StartDeserialization",
+                BindingFlags.Public | BindingFlags.Static,
+                Type.EmptyTypes);
```

Comment:
Worth caching (or caching a delegate created to it)?

Reply:
It would be even better to use UnsafeAccessor - no reflection, no delegates, no caching. UnsafeAccessor 
 shipped in 8.0 that is the current `NetCoreAppMinimum`.

Reply:
`StartDeserialization` returns a public type that isn't in the reference assembly; it looks like `UnsafeAccessor` requires an exact signature match.  So, unless there's an unsafe typeref I don't think we can use it here.

Reply:
`CreateDelegate` is similarly not an option, it refuses to loose-bind the return type (struct).

Slow invoke works, or we can add another method in corelib that does the boxing to IDisposable there so it can be bound with UnsafeAccessor.  I'm guessing we don't like that option, so I'm guessing we're sticking with slow-invoke.  But either way I've moved this to a different type, and am at least saving the MethodInfo.

Reply:
Should we rather get rid of this (it would be my preference), or turn it into a public API? 

The non-public dependencies of OOB nuget packages like this one are always factory for problems.